File: 1282. Group the People Given the  Group Size They Belong To/solution.cpp
class Solution {
public:
    vector<vector<int>> groupThePeople(vector<int>& groupSizes) {
        unordered_map<int, vector<int>> hashmap;
        vector<vector<int>> res;

        for (int i = 0; i < groupSizes.size(); i++){
            int key = groupSizes[i];
            if (hashmap.find(key) != hashmap.end()){
                hashmap[key].push_back(i);
            }
            else{
                hashmap[key] = {i};
            }
            if (hashmap[key].size() == key){
                res.push_back(hashmap[key]);
                hashmap.erase(key);
            }

        }


        return res;
        
    }
};

File: 1282. Group the People Given the  Group Size They Belong To/solution.py
class Solution(object):
    def groupThePeople(self, groupSizes):
        """
        :type groupSizes: List[int]
        :rtype: List[List[int]]
        """
        res = []
        hashmap = {}

        for i in range(len(groupSizes)):
            if groupSizes[i] in hashmap:
                hashmap[groupSizes[i]].append(i)

            else:
                hashmap[groupSizes[i]] = [i]

            if len(hashmap[groupSizes[i]]) == groupSizes[i]:
                res.append(hashmap[groupSizes[i]])
                del hashmap[groupSizes[i]]

        return res
            